### PR TITLE
Allow chatting with different providers

### PR DESCRIPTION
**Pull request by alberto98fx**
With this PR it's now possible to use also different models and providers included ollama for local execution: 
You can configure everything you want from `.env` (check example I supplied)

Providers supported are:
- OpenAI 
- Anthropic 
- Groq
- OpenRouter
- Ollama

Disclaimer: I am not a professional coder so it might be that I did not apply the best practices here... Any help or suggestion is welcome :)

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (ironman5366) [inline package-lock.json:1]: why do we need this lockfile at the top level?

File: api/ai.py [hunk 2]
```diff
@@ -1,9 +1,14 @@
+import os
 import time
 from datetime import datetime, timezone
 from invoke_types import InvocationRequest, Actor, LLMMessage
-from settings import MODEL, MODEL_KEY, MAX_TOKENS
+from settings import MODEL, MODEL_KEY, MAX_TOKENS, INFERENCE_SERVICE, API_KEY, OLLAMA_URL, GROQ_API_BASE, OPENROUTER_API_BASE
 import json
 import anthropic
+import openai
+from dotenv import load_dotenv
```
Comment (ironman5366) [inline api/ai.py:9]: nit: we should already have .env loaded bc we import `settings` which calls `load_dotenv`. doesn't really matter tho

**Review by ironman5366 — APPROVED** [review]
haven't looked at this too deeply but LGTM! thanks for the contribution. Clean up the couple small comments I left if you want, but regardless I'll merge this in in a few hours 👍

File: package-lock.json (hunk 1 above)
Comment (alberto98fx) [reply package-lock.json]: That is indeed a mistake! Thank you :)

File: api/ai.py (hunk 2 above)
Comment (alberto98fx) [reply api/ai.py]: good point!

**Comment by alberto98fx** [comment]
Hey @ironman5366 
I took the chance to also create a PoC Github action (https://github.com/alberto98fx/ai-murder-mystery-hackathon/actions/runs/10093046001/job/27907943930): for this iteration I commented out the health probe because for some reasons it does not want to give back the 200 status code and hence fails within the 30 seconds timeout. 
I will investigate why this happen when I have a little bit more time to dedicate. 
I added a small health check endpoint that maybe could be improved a little.

**Pull request merged**